pkg/environment/environment_provider.go

```diff
@@ -0,0 +1,55 @@
+package environment
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const defaultNamespace = "keptn"
+const secretPrefix = "locust"
+
+type EnvironmentProvider struct {
+	KubeAPI                v1.CoreV1Interface
+	KeptnNamespaceProvider StringSupplier
+}
+
+func NewEnvironmentProvider(kubeAPI v1.CoreV1Interface) *EnvironmentProvider {
+	return &EnvironmentProvider{
+		KubeAPI:                kubeAPI,
+		KeptnNamespaceProvider: envBasedStringSupplier("POD_NAMESPACE", defaultNamespace),
+	}
+}
+
+func (e EnvironmentProvider) PrepareEnvironment(project string, stage string, service string) []string {
+	environment := []string{}
+	secretName := fmt.Sprintf("%s-%s-%s-%s", secretPrefix, project, stage, service)
+	log.Printf("Prepare data of secret %s as environment", secretName)
+
+	secret, err := e.KubeAPI.Secrets(e.KeptnNamespaceProvider()).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		log.Printf("Unable to get secret %s: %s", secretName, err)
+		return environment
+	}
+
+	for key, value := range secret.Data {
+		environment = append(environment, fmt.Sprintf("%s=%s", key, string(value)))
```

Comment:
I think here we need to be careful and ensure that values that contain `"` and `=` also work.

E.g., what if my secret looks like this:
```
ABC="my key=value"
```

Reply:
I refactored the tests to make sure the generated environment is correct in the context of a command execution. Also added a test for more complex environment variables.